lisa/sut_orchestrator/azure/features.py

```diff
@@ -689,22 +690,38 @@ def setup_rdma(self) -> None:
         if len(devices) > 1:
             # upgrade waagent to latest version to resolve
             # multiple ib devices not getting ip address issue
-            waagent.upgrade_from_source()
+            waagent.upgrade_from_source("v2.9.0.4")
         # Update waagent.conf
         sed = self._node.tools[Sed]
-        sed.substitute(
-            regexp="# OS.EnableRDMA=y",
-            replacement="OS.EnableRDMA=y",
-            file="/etc/waagent.conf",
-            sudo=True,
-        )
-        sed.substitute(
-            regexp="# AutoUpdate.Enabled=y",
-            replacement="AutoUpdate.Enabled=y",
-            file="/etc/waagent.conf",
-            sudo=True,
-        )
-        waagent.restart()
+        if isinstance(self._node.os, CBLMariner):
+            sed.append(
+                text="OS.EnableRDMA=y",
+                file="/etc/waagent.conf",
+                sudo=True,
+            )
+        else:
+            sed.substitute(
```

Comment:
Those logic is still valid in Mariner, because the original string is not found.

Reply:
Fixed